api-service/src/controllers/DatasetRead/DatasetRead.ts

```diff
@@ -36,10 +36,19 @@ const datasetRead = async (req: Request, res: Response) => {
         throw obsrvError(dataset_id, "DATASET_NOT_FOUND", `Dataset with the given dataset_id:${dataset_id} not found`, "NOT_FOUND", 404);
     }
     if (dataset.connectors_config) {
-        dataset.connectors_config = dataset.connectors_config.map((connector: any) => ({
-            ...connector,
-            connector_config: JSON.parse(cipherService.decrypt(connector.connector_config))
-        }));
+        dataset.connectors_config = dataset?.connectors_config.map((connector: any) => {
```

Comment:
Move this to separate method this block of code

Reply:
Resolved